lib/infra/infra-stack.ts

```diff
@@ -0,0 +1,469 @@
+/* Copyright OpenSearch Contributors
+SPDX-License-Identifier: Apache-2.0
+
+The OpenSearch Contributors require contributions made to
+this file be licensed under the Apache-2.0 license or a
+compatible open source license. */
+
+import {
+  AmazonLinuxCpuType,
+  AmazonLinuxGeneration,
+  CloudFormationInit,
+  InitCommand,
+  InitElement,
+  InitPackage,
+  InstanceClass,
+  InstanceSize,
+  InstanceType,
+  ISecurityGroup,
+  IVpc,
+  MachineImage,
+  SubnetType,
+} from 'aws-cdk-lib/aws-ec2';
+import { ManagedPolicy, Role, ServicePrincipal } from 'aws-cdk-lib/aws-iam';
+import { AutoScalingGroup, BlockDeviceVolume, Signals } from 'aws-cdk-lib/aws-autoscaling';
+import { LogGroup, RetentionDays } from 'aws-cdk-lib/aws-logs';
+import {
+  CfnOutput, RemovalPolicy, Stack, StackProps, Tags,
+} from 'aws-cdk-lib';
+import { NetworkListener, NetworkLoadBalancer, Protocol } from 'aws-cdk-lib/aws-elasticloadbalancingv2';
+import { join } from 'path';
+import { readFileSync } from 'fs';
+import { dump, load } from 'js-yaml';
+import { CloudwatchAgent } from '../cloudwatch/cloudwatch-agent';
+import { nodeConfig } from '../opensearch-config/node-config';
+
+export interface infraProps extends StackProps{
+    readonly vpc: IVpc,
+    readonly securityGroup: ISecurityGroup,
+    readonly opensearchVersion: string,
+    readonly cpuArch: string,
+    readonly cpuType: AmazonLinuxCpuType,
+    readonly securityDisabled: boolean,
+    readonly minDistribution: boolean,
+    readonly distributionUrl: string,
+    readonly dashboardsUrl: string,
+    readonly singleNodeCluster: boolean,
+    readonly managerNodeCount: number,
+    readonly dataNodeCount: number,
+    readonly ingestNodeCount: number,
+    readonly clientNodeCount: number,
+    readonly mlNodeCount: number,
+    readonly jvmSysPropsString?: string
+}
+
+export class InfraStack extends Stack {
+  constructor(scope: Stack, id: string, props: infraProps) {
+    super(scope, id, props);
+    let opensearchListener: NetworkListener;
+    let dashboardsListener: NetworkListener;
+    let managerAsgCapacity: number;
+    let dataAsgCapacity: number;
+    let clientNodeAsg: AutoScalingGroup;
+    let seedConfig: string;
+    let hostType: InstanceType;
+
+    const clusterLogGroup = new LogGroup(this, 'opensearchLogGroup', {
+      logGroupName: 'opensearchLogGroup/opensearch.log',
+      retention: RetentionDays.ONE_MONTH,
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+
+    const instanceRole = new Role(this, 'instanceRole', {
+      roleName: 'OS-Instance-Role',
```

Comment:
remove role name as this will block users from deploying multiple clusters in same account. The cluster creation will fail saying resource already exists.